lang/csharp/src/apache/main/Protocol/Protocol.cs

```diff
@@ -147,8 +147,14 @@ private static Protocol Parse(JToken jtok)
                     messages.Add(message.Name, message);
                 }
             }
-
-            return new Protocol(name, space, doc, types, messages);
+            try
+            {
+                return new Protocol(name, space, doc, types, messages);
```

Comment:
I don't see any reason that this call would throw an exception. I think you can safely remove this `try`/`catch`.

Reply:
In ArraySchema.cs line 62
        private ArraySchema(Schema items, PropertyMap props) : base(Type.Array, props)
        {
            if (null == items) throw new ArgumentNullException("items");
            this.ItemSchema = items;
        }

Reply:
In Protocol.cs line 80

        public Protocol(string name, string space,
                        string doc, IEnumerable<Schema> types,
                        IDictionary<string,Message> messages)
        {
            if (string.IsNullOrEmpty(name)) throw new ArgumentNullException("name", "name cannot be null.");
            if (null == types) throw new ArgumentNullException("types", "types cannot be null.");
            if (null == messages) throw new ArgumentNullException("messages", "messages cannot be null.");

Reply:
done